<a href="https://colab.research.google.com/github/RRM2021/National-Land-Agency/blob/main/GESTION_UGT_ANT.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

AGENCIA NACIONAL DE TIERRAS ANT - UNIDAD DE GESTIÓN TERRITORIAL UGT  ANTIOQUIA

IDENTIFICACIÓN DE DUPLICIDADES Y DE VARIACIÓN EN LA INFORMACIÓN EN LA MATRIZ DE SEGUIMIENTO DE LA SUBDIRECCIÓN JURÍDICA DE LA UGT ANTIOQUIA

INGENIERO RICARDO RAMÍREZ - CIENTÍFICO DE DATOS - CONTRATISTA

OBTENCIÓN DE LOS DATOS

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving MATRIZ_DE_GESTION_UGT-ANTIOQUIA-2024 - CORTE DEL 3 DE ABRIL.xlsb to MATRIZ_DE_GESTION_UGT-ANTIOQUIA-2024 - CORTE DEL 3 DE ABRIL.xlsb


In [ ]:
!pip install pandas pyxlsb


In [ ]:
import pandas as pd
from pyxlsb import open_workbook as open_xlsb

# Replace 'path_to_file.xlsb' with the actual path to your uploaded XLSB file
df = []
with open_xlsb('/content/MATRIZ_DE_GESTION_UGT-ANTIOQUIA-2024 - CORTE DEL 3 DE ABRIL.xlsb') as wb:
    with wb.get_sheet(1) as sheet:
        for row in sheet.rows():
            df.append([item.v for item in row])

df = pd.DataFrame(df[1:], columns=df[0])


In [ ]:
df

MEMORANDO_DE_ENTREGA  FECHA_DE_ENTREGA MEMORANDO_SIG FECHA_SIG      ID  \
0             2.023310e+13           44985.0          None      None     0.0   
1             2.023310e+13           44985.0          None      None     1.0   
2             2.023310e+13           44985.0          None      None     2.0   
3             2.023310e+13           44985.0          None      None     3.0   
4             2.023310e+13           44985.0          None      None     4.0   
...                    ...               ...           ...       ...     ...   
3382                   NaN               NaN          None      None  3382.0   
3383                   NaN               NaN          None      None  3383.0   
3384                   NaN               NaN          None      None  3384.0   
3385                   NaN               NaN          None      None  3385.0   
3386                   NaN               NaN          None      None  3386.0   

     ESTRATEGIA CODIGO_SOLICITUD_SIG/FISO     EXPEDIENTE_ORFEO  \
0        REZAGO          51291000090002.0  202231004000204302E   
1        REZAGO          51291000090004.0  202231004000204308E   
2        REZAGO          51291000090010.0  202231004000204311E   
3        REZAGO          51291000090011.0  202231004000204319E   
4        REZAGO          51291000090017.0  202231004000204321E   
...         ...                       ...                  ...   
3382        FAM                  958776.0                 None   
3383        FAM                  958777.0                 None   
3384        FAM                  958778.0                 None   
3385        FAM                  976289.0                 None   
3386        FAM                 1282612.0                 None   

                NOMBRE_COMPLETO_SOLICITANTE CEDULA_SOLICITANTE  \
0                    RODRIGO TASCON SÁNCHEZ          6401149.0   
1                    ANA MILENA MAZO OSPINA         43688531.0   
2     EDILMA DEL SOCORRO ESPINOSA DE ZAPATA         21920060.0   
3                 MARÍA LUCÍA FLOREZ ZAPATA         39169543.0   
4                MARÍA STELLA MORALES DIOSA         39168929.0   
...                                     ...                ...   
3382                                   None               None   
3383                                   None               None   
3384                                   None               None   
3385                                   None               None   
3386                                   None               None   

     GENERO_SOLLICITANTE NOMBRE_COMPLETO_CONYUGE CEDULA_CONYUGE  \
0              MASCULINO                    None           None   
1               FEMENINO                    None           None   
2               FEMENINO                    None           None   
3               FEMENINO                    None           None   
4               FEMENINO                    None           None   
...                  ...                     ...            ...   
3382                None                    None           None   
3383                None                    None           None   
3384                None                    None           None   
3385                None                    None           None   
3386                None                    None           None   

     GENERO_CONYUGE NUMERO_DE_CONTACTO GENERO_TITULACION  \
0              None               None              None   
1              None               None              None   
2              None               None              None   
3              None               None              None   
4              None               None              None   
...             ...                ...               ...   
3382           None               None              None   
3383           None               None              None   
3384           None               None              None   
3385           None               None              No

ANÁLISIS EXPLORATORIO DE LOS DATOS

In [ ]:
df.columns


Index(['MEMORANDO_DE_ENTREGA', 'FECHA_DE_ENTREGA', 'MEMORANDO_SIG',
       'FECHA_SIG', 'ID', 'ESTRATEGIA', 'CONDIGO_QR',
       'CODIGO_SOLICITUD_SIG/FISO', 'EXPEDIENTE_ORFEO',
       'NOMBRE_COMPLETO_SOLICITANTE', 'CEDULA_SOLICITANTE',
       'GENERO_SOLLICITANTE', 'NOMBRE_COMPLETO_CONYUGE', 'CEDULA_CONYUGE',
       'GENERO_CONYUGE', 'NUMERO_DE_CONTACTO', 'GENERO_TITULACION',
       'PROPIETARIOS_TERCER_INTERESADO', 'ACREEDOR_HIPOTECA_TERCER_INTERESADO',
       'ACREEDOR_EMBARGO_TERCER_INTERESADO', 'COD_DEPARTAMENTO',
       'NOMBRE_DEPARTAMENTO', 'OCULTO', 'COD_MUNICIPIO', 'NOMBRE_MUNICIPIO',
       'VEREDA', 'DIRECCION/NOMBRE_PREDIO', 'NUMERO_PREDIAL_(30_DIGITOS)',
       'FMI', 'PREVIABILIDAD_JURIDICA', 'CATEGORIZACION_DE_PREDIAGNOSTICO',
       'GESTION_DOCUMENTAL_¿VERIFICO_LOS_SOPORTES_REQUERIDOS?', 'FECHA_DE_LPP',
       'FECHA_DE_ASIGNACION_POSTPROCESO_+_GDB_FINAL',
       'FECHA_DE_ENTREGA_POSTPROCESO_+_GDB_FINAL',
       'AREA_PREDIO_A_FORMALIZAR_(Ha)\n(0000,0000)',
       '

In [ ]:
df.shape

(3414, 77)

In [ ]:
# Filter using square brackets []
UGT_df = df[df['COD_DEPARTAMENTO'] == '05']
print("Result dataframe:")
pd.set_option('display.max_columns', None) # Set the display option

UGT_df.info()


Result dataframe:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3414 entries, 0 to 3413
Data columns (total 77 columns):
 #   Column                                                  Non-Null Count  Dtype  
---  ------                                                  --------------  -----  
 0   MEMORANDO_DE_ENTREGA                                    2864 non-null   float64
 1   FECHA_DE_ENTREGA                                        2864 non-null   float64
 2   MEMORANDO_SIG                                           0 non-null      object 
 3   FECHA_SIG                                               0 non-null      object 
 4   ID                                                      3414 non-null   float64
 5   ESTRATEGIA                                              3067 non-null   object 
 6   CONDIGO_QR                                              0 non-null      object 
 7   CODIGO_SOLICITUD_SIG/FISO                               3150 non-null   object 
 8   EXPEDIENTE_ORFEO    

In [ ]:
UGT_df.shape

(3414, 77)

*La matriz consta de 3414 filas y 77 columnas*

In [ ]:
# Initialize dictionaries to store counts
null_counts = {}
not_null_counts = {}

# Iterate over columns
for col in df.columns:
    null_counts[col] = df[col].isnull().sum()
    not_null_counts[col] = df[col].notnull().sum()

# Create DataFrame from dictionaries
null_counts_df = pd.DataFrame.from_dict(null_counts, orient='index', columns=['Null Count'])
not_null_counts_df = pd.DataFrame.from_dict(not_null_counts, orient='index', columns=['Not Null Count'])

# Concatenate DataFrames
result_df = pd.concat([null_counts_df, not_null_counts_df], axis=1)

print("Null and Not Null Value Counts:")
print(result_df)
result_df


Null and Not Null Value Counts:
                                             Null Count  Not Null Count
MEMORANDO_DE_ENTREGA                                550            2864
FECHA_DE_ENTREGA                                    550            2864
MEMORANDO_SIG                                      3414               0
FECHA_SIG                                          3414               0
ID                                                    0            3414
...                                                 ...             ...
TITULO_ENTREGADO                                   3414               0
FECHA_PROGRAMADA_PARA_LA_ENTREGA_DEL_TITULO        3414               0
FECHA_DE_ENTREGA_EFECTIVA_DEL_TITULO               3414               0
REPORTADO_A_BMT                                    3414               0
FECHA_DE_REGISTRO_EN_BMT                           3414               0

[77 rows x 2 columns]


Null Count  Not Null Count
MEMORANDO_DE_ENTREGA                                550            2864
FECHA_DE_ENTREGA                                    550            2864
MEMORANDO_SIG                                      3414               0
FECHA_SIG                                          3414               0
ID                                                    0            3414
...                                                 ...             ...
TITULO_ENTREGADO                                   3414               0
FECHA_PROGRAMADA_PARA_LA_ENTREGA_DEL_TITULO        3414               0
FECHA_DE_ENTREGA_EFECTIVA_DEL_TITULO               3414               0
REPORTADO_A_BMT                                    3414               0
FECHA_DE_REGISTRO_EN_BMT                           3414               0

[77 rows x 2 columns]

In [ ]:
# Assuming you have already created the 'result_df' DataFrame
result_df.to_excel('result_df.xlsx', index=True)  # Save to an Excel file
from google.colab import files
files.download('result_df.xlsx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

*IDENTIFICACIÓN DE DATOS NULOS (True) y NO NULOS (False) EN LAS COLUMNAS*

In [ ]:
# Identificación de datos nulos
missing_data = df.isnull()
missing_data.head(5)

MEMORANDO_DE_ENTREGA  FECHA_DE_ENTREGA  MEMORANDO_SIG  FECHA_SIG     ID  \
0                 False             False           True       True  False   
1                 False             False           True       True  False   
2                 False             False           True       True  False   
3                 False             False           True       True  False   
4                 False             False           True       True  False   

   ESTRATEGIA  CONDIGO_QR  CODIGO_SOLICITUD_SIG/FISO  EXPEDIENTE_ORFEO  \
0       False        True                      False             False   
1       False        True                      False             False   
2       False        True                      False             False   
3       False        True                      False             False   
4       False        True                      False             False   

   NOMBRE_COMPLETO_SOLICITANTE  CEDULA_SOLICITANTE  GENERO_SOLLICITANTE  \
0                        False               False                False   
1                        False               False                False   
2                        False               False                False   
3                        False               False                False   
4                        False               False                False   

   NOMBRE_COMPLETO_CONYUGE  CEDULA_CONYUGE  GENERO_CONYUGE  \
0                     True            True            True   
1                     True            True            True   
2                     True            True            True   
3                     True            True            True   
4                     True            True            True   

   NUMERO_DE_CONTACTO  GENERO_TITULACION  PROPIETARIOS_TERCER_INTERESADO  \
0                True               True                            True   
1                True               True                            True   
2                True               True                            True   
3                True               True                            True   
4                True               True                            True   

   ACREEDOR_HIPOTECA_TERCER_INTERESADO  ACREEDOR_EMBARGO_TERCER_INTERESADO  \
0                                 True                                True   
1                                 True                                True   
2                                 True                                True   
3                                 True                                True   
4                                 True                                True   

   COD_DEPARTAMENTO  NOMBRE_DEPARTAMENTO  OCULTO  COD_MUNICIPIO  \
0             False                False   False          False   
1             False                False   False          False   
2             False                False   False          False   
3             False                False   False          False   
4             False                False   False          False   

   NOMBRE_MUNICIPIO  VEREDA  DIRECCION/NOMBRE_PREDIO  \
0             False    True                     True   
1             False    True                     True   
2             False    True                     True   
3             False    True                     True   
4             False    True                     True   

   NUMERO_PREDIAL_(30_DIGITOS)   FMI  PREVIABILIDAD_JURIDICA  \
0                         True  True                    True   
1                         True  True                    True   
2                         True  True                    True   
3                         True  True                    True   
4                         True  True                    True   

   CATEGORIZACION_DE_PREDIAGNOSTICO  \
0                              True   
1                              True   
2                              True   
3                              True   
4                  

In [ ]:
for column in missing_data.columns.values.tolist():
    print(column)
    print (missing_data[column].value_counts())
    print("")

MEMORANDO_DE_ENTREGA
MEMORANDO_DE_ENTREGA
False    2864
True      550
Name: count, dtype: int64

FECHA_DE_ENTREGA
FECHA_DE_ENTREGA
False    2864
True      550
Name: count, dtype: int64

MEMORANDO_SIG
MEMORANDO_SIG
True    3414
Name: count, dtype: int64

FECHA_SIG
FECHA_SIG
True    3414
Name: count, dtype: int64

ID
ID
False    3414
Name: count, dtype: int64

ESTRATEGIA
ESTRATEGIA
False    3067
True      347
Name: count, dtype: int64

CONDIGO_QR
CONDIGO_QR
True    3414
Name: count, dtype: int64

CODIGO_SOLICITUD_SIG/FISO
CODIGO_SOLICITUD_SIG/FISO
False    3150
True      264
Name: count, dtype: int64

EXPEDIENTE_ORFEO
EXPEDIENTE_ORFEO
True     2239
False    1175
Name: count, dtype: int64

NOMBRE_COMPLETO_SOLICITANTE
NOMBRE_COMPLETO_SOLICITANTE
False    2591
True      823
Name: count, dtype: int64

CEDULA_SOLICITANTE
CEDULA_SOLICITANTE
False    2381
True     1033
Name: count, dtype: int64

GENERO_SOLLICITANTE
GENERO_SOLLICITANTE
False    2813
True      601
Name: count, dtype: int64

NOMBR

IDENTIFICACIÓN DE DATOS DUPLICADOS

In [ ]:
duplicate_name_rows = df[df.duplicated(subset=['CODIGO_SOLICITUD_SIG/FISO'])]
print("Duplicate Rows based on 'CODIGO_SOLICITUD_SIG/FISO':")
duplicate_name_rows


Duplicate Rows based on 'CODIGO_SOLICITUD_SIG/FISO':


MEMORANDO_DE_ENTREGA  FECHA_DE_ENTREGA MEMORANDO_SIG FECHA_SIG      ID  \
2660          2.023310e+14           45174.0          None      None  2660.0   
2661          2.023310e+14           45174.0          None      None  2661.0   
2662          2.023310e+14           45174.0          None      None  2662.0   
2663          2.023310e+14           45174.0          None      None  2663.0   
2664          2.023310e+14           45174.0          None      None  2664.0   
...                    ...               ...           ...       ...     ...   
3348                   NaN               NaN          None      None  3348.0   
3350                   NaN               NaN          None      None  3350.0   
3351                   NaN               NaN          None      None  3351.0   
3354                   NaN               NaN          None      None  3354.0   
3355                   NaN               NaN          None      None  3355.0   

     ESTRATEGIA CONDIGO_QR CODIGO_SOLICITUD_SIG/FISO     EXPEDIENTE_ORFEO  \
2660       None       None                      None  202231004000202495E   
2661       None       None                      None  202231004000202498E   
2662       None       None                      None  202231004000202500E   
2663       None       None                      None  202231004000202502E   
2664       None       None                      None  202231004000202504E   
...         ...        ...                       ...                  ...   
3348        FAM       None                      None  202331004000201293E   
3350        FAM       None                      None  202331004000201236E   
3351        FAM       None                      None  202131004000201186E   
3354        FAM       None                      None  202331004000200933E   
3355        FAM       None                      None  202331004000200966E   

      NOMBRE_COMPLETO_SOLICITANTE CEDULA_SOLICITANTE GENERO_SOLLICITANTE  \
2660                         None               None                None   
2661                         None               None                None   
2662                         None               None                None   
2663                         None               None                None   
2664                         None               None                None   
...                           ...                ...                 ...   
3348   NOELIA DE JESUS HERNANDEZ          34505795.0            FEMENINO   
3350    ILLER YASMANIN MAZO OCHOA       1040363522.0           MASCULINO   
3351  YULIANA  CIFUENTES CASILIMA       1022332559.0            FEMENINO   
3354   NANCY LILIANA BORJA GUERRA       1017138774.0            FEMENINO   
3355      WILDERM  MACIAS RAIGOSA          8436425.0           MASCULINO   

     NOMBRE_COMPLETO_CONYUGE CEDULA_CONYUGE GENERO_CONYUGE NUMERO_DE_CONTACTO  \
2660                    None           None           None               None   
2661                    None           None           None               None   
2662                    None           None           None               None   
2663                    None           None           None               None   
2664                    None           None           None               None   
...                      ...            ...            ...                ...   
3348                    None           None           None               None   
3350                    None           None           None               None   
3351                    None           None           None               None   
3354                    None           None           None               None   
3355                    None           None           None               None   

     GENERO_TITULACION PROPIETARIOS_TERCER_INTERESADO  \
2660              None                           None   
2661              None                           None   
2662              None                           None   
2663  

In [ ]:
duplicate_name_rows.shape

(263, 77)

*Hay 263 filas con valores repetidos en la columna CODIGO_SOLICITUD_SIG/FISO; se incluyen filas con valores nulos o vacíos.*

In [ ]:
null_CODIGO_rows = df[~df['CODIGO_SOLICITUD_SIG/FISO'].isna()]
print("Rows without NaN values in 'CODIGO':")
print(null_CODIGO_rows.shape)
null_CODIGO_rows


Rows without NaN values in 'CODIGO':
(3150, 77)


MEMORANDO_DE_ENTREGA  FECHA_DE_ENTREGA MEMORANDO_SIG FECHA_SIG      ID  \
0             2.023310e+13           44985.0          None      None     0.0   
1             2.023310e+13           44985.0          None      None     1.0   
2             2.023310e+13           44985.0          None      None     2.0   
3             2.023310e+13           44985.0          None      None     3.0   
4             2.023310e+13           44985.0          None      None     4.0   
...                    ...               ...           ...       ...     ...   
3409                   NaN               NaN          None      None  3409.0   
3410                   NaN               NaN          None      None  3410.0   
3411                   NaN               NaN          None      None  3411.0   
3412                   NaN               NaN          None      None  3412.0   
3413                   NaN               NaN          None      None  3413.0   

                  ESTRATEGIA CONDIGO_QR CODIGO_SOLICITUD_SIG/FISO  \
0                     REZAGO       None          51291000090002.0   
1                     REZAGO       None          51291000090004.0   
2                     REZAGO       None          51291000090010.0   
3                     REZAGO       None          51291000090011.0   
4                     REZAGO       None          51291000090017.0   
...                      ...        ...                       ...   
3409  SOLICITUD POR WHATSAPP       None                   23514.0   
3410                     FAM       None                  589792.0   
3411                     FAM       None                 1307534.0   
3412                     FAM       None                 1307539.0   
3413                     FAM       None                 1414045.0   

          EXPEDIENTE_ORFEO            NOMBRE_COMPLETO_SOLICITANTE  \
0      202231004000204302E                 RODRIGO TASCON SÁNCHEZ   
1      202231004000204308E                 ANA MILENA MAZO OSPINA   
2      202231004000204311E  EDILMA DEL SOCORRO ESPINOSA DE ZAPATA   
3      202231004000204319E              MARÍA LUCÍA FLOREZ ZAPATA   
4      202231004000204321E             MARÍA STELLA MORALES DIOSA   
...                    ...                                    ...   
3409                  None                                   None   
3410  2023220106998279420E        CARLOS HUMBERTO GONZALEZ PATIÑO   
3411  2023220106998757916E             FRANCISCO GOMEZ TORDECILLA   
3412  2023220106998757921E             NELSY ESTER MADERA PALACIO   
3413  2023220106998855531E                 RAMIRO RUIZ BLANQUICET   

     CEDULA_SOLICITANTE GENERO_SOLLICITANTE NOMBRE_COMPLETO_CONYUGE  \
0             6401149.0           MASCULINO                    None   
1            43688531.0            FEMENINO                    None   
2            21920060.0            FEMENINO                    None   
3            39169543.0            FEMENINO                    None   
4            39168929.0            FEMENINO                    None   
...                 ...                 ...                     ...   
3409               None                None                    None   
3410            7487884           MASCULINO                    None   
3411           98613036           MASCULINO                    None   
3412           32201326            FEMENINO                    None   
3413            8424277           MASCULINO                    None   

     CEDULA_CONYUGE GENERO_CONYUGE NUMERO_DE_CONTACTO GENERO_TITULACION  \
0              None           None               None              None   
1              None           None               None              None   
2              None           None               None              None   
3              None           None               None              None   
4              None           None               None              None   
...             ...            ...                ...               ...   
3409     

*Hay 3150 filas con valores en la columna CODIGO_SOLICITUD_SIG/FISO*

In [ ]:
null_CODIGO_rows2 = df[df['CODIGO_SOLICITUD_SIG/FISO'].isna()]
print("Rows with NaN values in 'CODIGO_SOLICITUD_SIG/FISO':")
print(null_CODIGO_rows2.shape)
null_CODIGO_rows2

Rows with NaN values in 'CODIGO_SOLICITUD_SIG/FISO':
(264, 77)


MEMORANDO_DE_ENTREGA  FECHA_DE_ENTREGA MEMORANDO_SIG FECHA_SIG      ID  \
2659          2.023310e+14           45174.0          None      None  2659.0   
2660          2.023310e+14           45174.0          None      None  2660.0   
2661          2.023310e+14           45174.0          None      None  2661.0   
2662          2.023310e+14           45174.0          None      None  2662.0   
2663          2.023310e+14           45174.0          None      None  2663.0   
...                    ...               ...           ...       ...     ...   
3348                   NaN               NaN          None      None  3348.0   
3350                   NaN               NaN          None      None  3350.0   
3351                   NaN               NaN          None      None  3351.0   
3354                   NaN               NaN          None      None  3354.0   
3355                   NaN               NaN          None      None  3355.0   

     ESTRATEGIA CONDIGO_QR CODIGO_SOLICITUD_SIG/FISO     EXPEDIENTE_ORFEO  \
2659       None       None                      None  202231004000200577E   
2660       None       None                      None  202231004000202495E   
2661       None       None                      None  202231004000202498E   
2662       None       None                      None  202231004000202500E   
2663       None       None                      None  202231004000202502E   
...         ...        ...                       ...                  ...   
3348        FAM       None                      None  202331004000201293E   
3350        FAM       None                      None  202331004000201236E   
3351        FAM       None                      None  202131004000201186E   
3354        FAM       None                      None  202331004000200933E   
3355        FAM       None                      None  202331004000200966E   

      NOMBRE_COMPLETO_SOLICITANTE CEDULA_SOLICITANTE GENERO_SOLLICITANTE  \
2659                         None               None                None   
2660                         None               None                None   
2661                         None               None                None   
2662                         None               None                None   
2663                         None               None                None   
...                           ...                ...                 ...   
3348   NOELIA DE JESUS HERNANDEZ          34505795.0            FEMENINO   
3350    ILLER YASMANIN MAZO OCHOA       1040363522.0           MASCULINO   
3351  YULIANA  CIFUENTES CASILIMA       1022332559.0            FEMENINO   
3354   NANCY LILIANA BORJA GUERRA       1017138774.0            FEMENINO   
3355      WILDERM  MACIAS RAIGOSA          8436425.0           MASCULINO   

     NOMBRE_COMPLETO_CONYUGE CEDULA_CONYUGE GENERO_CONYUGE NUMERO_DE_CONTACTO  \
2659                    None           None           None               None   
2660                    None           None           None               None   
2661                    None           None           None               None   
2662                    None           None           None               None   
2663                    None           None           None               None   
...                      ...            ...            ...                ...   
3348                    None           None           None               None   
3350                    None           None           None               None   
3351                    None           None           None               None   
3354                    None           None           None               None   
3355                    None           None           None               None   

     GENERO_TITULACION PROPIETARIOS_TERCER_INTERESADO  \
2659              None                           None   
2660              None                           None   
2661              None                           None   
2662  

*Hay 264 filas con valores nulos o celdas vacías, en la columna CODIGO_SOLICITUD_SIG/FISO*

In [ ]:
duplicate_name_rows2 = null_CODIGO_rows[null_CODIGO_rows.duplicated(subset=['CODIGO_SOLICITUD_SIG/FISO'])]
print("Duplicate Rows2 based on 'Name':")
print(duplicate_name_rows2.shape)
duplicate_name_rows2

Duplicate Rows2 based on 'Name':
(0, 76)


Empty DataFrame
Columns: [MEMORANDO_DE_ENTREGA, FECHA_DE_ENTREGA, MEMORANDO_SIG, FECHA_SIG, ID, ESTRATEGIA, CODIGO_SOLICITUD_SIG/FISO, EXPEDIENTE_ORFEO, NOMBRE_COMPLETO_SOLICITANTE, CEDULA_SOLICITANTE, GENERO_SOLLICITANTE, NOMBRE_COMPLETO_CONYUGE, CEDULA_CONYUGE, GENERO_CONYUGE, NUMERO_DE_CONTACTO, GENERO_TITULACION, PROPIETARIOS_TERCER_INTERESADO, ACREEDOR_HIPOTECA_TERCER_INTERESADO, ACREEDOR_EMBARGO_TERCER_INTERESADO, COD_DEPARTAMENTO, NOMBRE_DEPARTAMENTO, OCULTO, COD_MUNICIPIO, NOMBRE_MUNICIPIO, VEREDA, DIRECCION/NOMBRE_PREDIO, NUMERO_PREDIAL_(30_DIGITOS), FMI, PREVIABILIDAD_JURIDICA, CATEGORIZACION_DE_PREDIAGNOSTICO, GESTION_DOCUMENTAL_¿VERIFICO_LOS_SOPORTES_REQUERIDOS?, FECHA_DE_LPP, FECHA_DE_ASIGNACION_POSTPROCESO_+_GDB_FINAL, FECHA_DE_ENTREGA_POSTPROCESO_+_GDB_FINAL, AREA_PREDIO_A_FORMALIZAR_(Ha)
(0000,0000), FECHA_DE_ASIGNACION_ELABORACION_ITJ_Y_PLANO, ESTADO_ITJ, FECHA_ENTREGA_A_COMPONENTE_JURIDICO, RUTA_DE_GESTION_JURIDICA, FECHA_DE_ASIGNACION, EXCLUSIONES_(APLICA_SOLO_PARA_EL_ESTADO_"INVIABILIDAD", RADICADO_DE_ORFEO_DEL_ACTO_DEL_INICIO/ARCHIVO, FECHA_DE_ACTO_DEL_INICIO/ARCHIVO, NOTIFICACIONES_Y_COMUNICACIONES_INICIO, AUDIENCIA, FECHA_DE_AUDIENCIA, COD_ORIP, NOMBRE_ORIP, FECHA_DE_ENVIO_A_LA_ORIP, TURNO, ESTADO, ESTADO_PU, CATEGORIA_DE_INVIABILIDAD, RADICADO_DE_ORFEO_DEL_RECURSO, ACTO_ADMINISTRATIVO_DE_PERIODO_PROBATORIO, FECHA_ACTO_ADMINISTRATIVO_DE_PERIODO_PROBATORIO, ITJD, TIPO_DE_ACTO_ADMINISTRATIVO_DE_CIERRE, FECHA_DE_ASIGNACION2, EXCLUSIONES2, RADICADO_DEL_ACTO_DEL_CIERRE, FECHA_ACTO_DE_CIERRE, RADICADO_DE_ORFEO_DEL_RECURSO2, RECURSO_RESUELTO, FECHA_RADICADO_EN_ORIP, RADICADO_MEDIO_ELECTRONICO_O_FISICO, TURNO_ASIGNADO_EN_LA_ORIP, ESTADO2, CAUSAL_NOTA_DEVOLUTIVA, FMI_NUEVO, FECHA_FMI_NUEVO, TITULO_ENTREGADO, FECHA_PROGRAMADA_PARA_LA_ENTREGA_DEL_TITULO, FECHA_DE_ENTREGA_EFECTIVA_DEL_TITULO, REPORTADO_A_BMT, FECHA_DE_REGISTRO_EN_BMT]
Index: []

In [ ]:
import pandas as pd

# Initialize an empty dictionary to store column-wise duplicate counts
column_duplicate_counts = {}

# Iterate over each column
for col in df.columns:
    # Count the duplicates for the current column
    duplicate_count = len(df) - len(df.drop_duplicates(subset=col))
    column_duplicate_counts[col] = duplicate_count

# Create a DataFrame from the dictionary
result_df3 = pd.DataFrame(column_duplicate_counts.items(), columns=['Column', 'Duplicate Count'])

print("Duplicate counts based on each column:")

# Assuming your DataFrame is named 'df'
pd.set_option('display.max_rows', None)  # Set to display all rows
print(result_df3)
#result_df.head(76)

Duplicate counts based on each column:
                                               Column  Duplicate Count
0                                MEMORANDO_DE_ENTREGA             3404
1                                    FECHA_DE_ENTREGA             3406
2                                       MEMORANDO_SIG             3413
3                                           FECHA_SIG             3413
4                                                  ID                0
5                                          ESTRATEGIA             3409
6                                          CONDIGO_QR             3413
7                           CODIGO_SOLICITUD_SIG/FISO              263
8                                    EXPEDIENTE_ORFEO             2238
9                         NOMBRE_COMPLETO_SOLICITANTE             1056
10                                 CEDULA_SOLICITANTE             1242
11                                GENERO_SOLLICITANTE             3410
12                            NOMBRE_C

In [ ]:
# Identificación de datos nulos
missing_data = df.isnull()
missing_data.head(5)

MEMORANDO_DE_ENTREGA  FECHA_DE_ENTREGA  MEMORANDO_SIG  FECHA_SIG     ID  \
0                 False             False           True       True  False   
1                 False             False           True       True  False   
2                 False             False           True       True  False   
3                 False             False           True       True  False   
4                 False             False           True       True  False   

   ESTRATEGIA  CONDIGO_QR  CODIGO_SOLICITUD_SIG/FISO  EXPEDIENTE_ORFEO  \
0       False        True                      False             False   
1       False        True                      False             False   
2       False        True                      False             False   
3       False        True                      False             False   
4       False        True                      False             False   

   NOMBRE_COMPLETO_SOLICITANTE  CEDULA_SOLICITANTE  GENERO_SOLLICITANTE  \
0                        False               False                False   
1                        False               False                False   
2                        False               False                False   
3                        False               False                False   
4                        False               False                False   

   NOMBRE_COMPLETO_CONYUGE  CEDULA_CONYUGE  GENERO_CONYUGE  \
0                     True            True            True   
1                     True            True            True   
2                     True            True            True   
3                     True            True            True   
4                     True            True            True   

   NUMERO_DE_CONTACTO  GENERO_TITULACION  PROPIETARIOS_TERCER_INTERESADO  \
0                True               True                            True   
1                True               True                            True   
2                True               True                            True   
3                True               True                            True   
4                True               True                            True   

   ACREEDOR_HIPOTECA_TERCER_INTERESADO  ACREEDOR_EMBARGO_TERCER_INTERESADO  \
0                                 True                                True   
1                                 True                                True   
2                                 True                                True   
3                                 True                                True   
4                                 True                                True   

   COD_DEPARTAMENTO  NOMBRE_DEPARTAMENTO  OCULTO  COD_MUNICIPIO  \
0             False                False   False          False   
1             False                False   False          False   
2             False                False   False          False   
3             False                False   False          False   
4             False                False   False          False   

   NOMBRE_MUNICIPIO  VEREDA  DIRECCION/NOMBRE_PREDIO  \
0             False    True                     True   
1             False    True                     True   
2             False    True                     True   
3             False    True                     True   
4             False    True                     True   

   NUMERO_PREDIAL_(30_DIGITOS)   FMI  PREVIABILIDAD_JURIDICA  \
0                         True  True                    True   
1                         True  True                    True   
2                         True  True                    True   
3                         True  True                    True   
4                         True  True                    True   

   CATEGORIZACION_DE_PREDIAGNOSTICO  \
0                              True   
1                              True   
2                              True   
3                              True   
4                  

In [ ]:
for column in missing_data.columns.values.tolist():
    print(column)
    print (missing_data[column].value_counts())
    print("")

MEMORANDO_DE_ENTREGA
MEMORANDO_DE_ENTREGA
False    2864
True      550
Name: count, dtype: int64

FECHA_DE_ENTREGA
FECHA_DE_ENTREGA
False    2864
True      550
Name: count, dtype: int64

MEMORANDO_SIG
MEMORANDO_SIG
True    3414
Name: count, dtype: int64

FECHA_SIG
FECHA_SIG
True    3414
Name: count, dtype: int64

ID
ID
False    3414
Name: count, dtype: int64

ESTRATEGIA
ESTRATEGIA
False    3067
True      347
Name: count, dtype: int64

CONDIGO_QR
CONDIGO_QR
True    3414
Name: count, dtype: int64

CODIGO_SOLICITUD_SIG/FISO
CODIGO_SOLICITUD_SIG/FISO
False    3150
True      264
Name: count, dtype: int64

EXPEDIENTE_ORFEO
EXPEDIENTE_ORFEO
True     2239
False    1175
Name: count, dtype: int64

NOMBRE_COMPLETO_SOLICITANTE
NOMBRE_COMPLETO_SOLICITANTE
False    2591
True      823
Name: count, dtype: int64

CEDULA_SOLICITANTE
CEDULA_SOLICITANTE
False    2381
True     1033
Name: count, dtype: int64

GENERO_SOLLICITANTE
GENERO_SOLLICITANTE
False    2813
True      601
Name: count, dtype: int64

NOMBR

In [ ]:
import pandas as pd

# Initialize an empty dictionary to store value-wise duplicate counts
value_duplicate_counts = {}

# Iterate over each column
for col in df.columns:
    # Count the duplicates for the current column (excluding null values)
    duplicate_count = len(df.dropna(subset=[col])) - len(df.drop_duplicates(subset=col, keep=False))
    value_duplicate_counts[col] = duplicate_count

# Create a DataFrame from the dictionary
result_df2 = pd.DataFrame(value_duplicate_counts.items(), columns=['Column', 'Duplicate Count'])

print("Value-wise duplicate counts for each column (excluding null values):")
print(result_df2)
result_df2


Value-wise duplicate counts for each column (excluding null values):
                                               Column  Duplicate Count
0                                MEMORANDO_DE_ENTREGA             2864
1                                    FECHA_DE_ENTREGA             2864
2                                       MEMORANDO_SIG                0
3                                           FECHA_SIG                0
4                                                  ID                0
5                                          ESTRATEGIA             3067
6                                          CONDIGO_QR                0
7                           CODIGO_SOLICITUD_SIG/FISO                0
8                                    EXPEDIENTE_ORFEO                0
9                         NOMBRE_COMPLETO_SOLICITANTE              405
10                                 CEDULA_SOLICITANTE              389
11                                GENERO_SOLLICITANTE             2813
12      

Column  Duplicate Count
0                                MEMORANDO_DE_ENTREGA             2864
1                                    FECHA_DE_ENTREGA             2864
2                                       MEMORANDO_SIG                0
3                                           FECHA_SIG                0
4                                                  ID                0
5                                          ESTRATEGIA             3067
6                                          CONDIGO_QR                0
7                           CODIGO_SOLICITUD_SIG/FISO                0
8                                    EXPEDIENTE_ORFEO                0
9                         NOMBRE_COMPLETO_SOLICITANTE              405
10                                 CEDULA_SOLICITANTE              389
11                                GENERO_SOLLICITANTE             2813
12                            NOMBRE_COMPLETO_CONYUGE               33
13                                     CEDULA_CONYUGE               34
14                                     GENERO_CONYUGE              173
15                                 NUMERO_DE_CONTACTO                0
16                                  GENERO_TITULACION              406
17                     PROPIETARIOS_TERCER_INTERESADO              164
18                ACREEDOR_HIPOTECA_TERCER_INTERESADO                5
19                 ACREEDOR_EMBARGO_TERCER_INTERESADO                2
20                                   COD_DEPARTAMENTO             3414
21                                NOMBRE_DEPARTAMENTO             3414
22                                             OCULTO             3414
23                                      COD_MUNICIPIO             3413
24                                   NOMBRE_MUNICIPIO             3413
25                                             VEREDA              582
26                            DIRECCION/NOMBRE_PREDIO              282
27                        NUMERO_PREDIAL_(30_DIGITOS)              149
28                                                FMI              584
29                             PREVIABILIDAD_JURIDICA              864
30                   CATEGORIZACION_DE_PREDIAGNOSTICO              735
31  GESTION_DOCUMENTAL_¿VERIFICO_LOS_SOPORTES_REQU...               10
32                                       FECHA_DE_LPP              136
33        FECHA_DE_ASIGNACION_POSTPROCESO_+_GDB_FINAL              135
34           FECHA_DE_ENTREGA_POSTPROCESO_+_GDB_FINAL              137
35         AREA_PREDIO_A_FORMALIZAR_(Ha)\n(0000,0000)               44
36        FECHA_DE_ASIGNACION_ELABORACION_ITJ_Y_PLANO              134
37                                         ESTADO_ITJ              952
38                FECHA_ENTREGA_A_COMPONENTE_JURIDICO              134
39                           RUTA_DE_GESTION_JURIDICA              875
40                                FECHA_DE_ASIGNACION               38
41  EXCLUSIONES_(APLICA_SOLO_PARA_EL_ESTADO_"INVIA...              256
42      RADICADO_DE_ORFEO_DEL_ACTO_DEL_INICIO/ARCHIVO                0
43                   FECHA_DE_ACTO_DEL_INICIO/ARCHIVO              549
44             NOTIFICACIONES_Y_COMUNICACIONES_INICIO              457
45                                          AUDIENCIA                0
46                                 FECHA_DE_AUDIENCIA                0
47                                           COD_ORIP             3414
48                                        NOMBRE_ORIP             3414
49                           FECHA_DE_ENVIO_A_LA_ORIP               43
50                                              TURNO                0
51                                             ESTADO               41
52                                          ESTADO_PU             3414
53                          CATEGORIA_DE_INVIABILIDAD              256
54                      RADICADO_DE_ORFEO_DEL_RECURSO                0
55          ACTO_ADMINISTRATIVO_DE_PERIODO_PROBATORIO                0


In [ ]:
# Assuming you have already created the 'result_df' DataFrame
result_df2.to_excel('result_df2.xlsx', index=False)  # Save to an Excel file


In [ ]:
from google.colab import files
files.download('result_df2.xlsx')


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
# Assuming your DataFrame is named 'df' and the column of interest is 'Column_Name'
duplicated_rows = df[df.duplicated('NOMBRE_COMPLETO_SOLICITANTE', keep=False) & ~df['NOMBRE_COMPLETO_SOLICITANTE'].isnull()]
print("Rows with duplicated values (excluding nulls):")
#print(duplicated_rows)
duplicated_rows

Rows with duplicated values (excluding nulls):


MEMORANDO_DE_ENTREGA  FECHA_DE_ENTREGA MEMORANDO_SIG FECHA_SIG      ID  \
7             2.023310e+13           44985.0          None      None     7.0   
60            2.023310e+13           44985.0          None      None    60.0   
81            2.023310e+13           44985.0          None      None    81.0   
85            2.023310e+13           44985.0          None      None    85.0   
119           2.023310e+13           44985.0          None      None   119.0   
123           2.023310e+13           44985.0          None      None   123.0   
124           2.023310e+13           44985.0          None      None   124.0   
125           2.023310e+13           44985.0          None      None   125.0   
207           2.023310e+13           44985.0          None      None   207.0   
209           2.023310e+13           44985.0          None      None   209.0   
219           2.023310e+13           44985.0          None      None   219.0   
220           2.023310e+13           44985.0          None      None   220.0   
221           2.023310e+13           44985.0          None      None   221.0   
223           2.023310e+13           44985.0          None      None   223.0   
224           2.023310e+13           44985.0          None      None   224.0   
227           2.023310e+13           44985.0          None      None   227.0   
228           2.023310e+13           44985.0          None      None   228.0   
251           2.023310e+13           44985.0          None      None   251.0   
253           2.023310e+13           44985.0          None      None   253.0   
265           2.023310e+13           44985.0          None      None   265.0   
266           2.023310e+13           44985.0          None      None   266.0   
267           2.023310e+13           44985.0          None      None   267.0   
444           2.023310e+13           44985.0          None      None   444.0   
445           2.023310e+13           44985.0          None      None   445.0   
522           2.023310e+13           44985.0          None      None   522.0   
553           2.023310e+13           44985.0          None      None   553.0   
556           2.023310e+13           44985.0          None      None   556.0   
561           2.023310e+13           44985.0          None      None   561.0   
583           2.023310e+13           44985.0          None      None   583.0   
584           2.023310e+13           44985.0          None      None   584.0   
586           2.023310e+13           44985.0          None      None   586.0   
587           2.023310e+13           44985.0          None      None   587.0   
592           2.023310e+13           44985.0          None      None   592.0   
593           2.023310e+13           44985.0          None      None   593.0   
600           2.023310e+13           44985.0          None      None   600.0   
601           2.023310e+13           44985.0          None      None   601.0   
616           2.023310e+13           44985.0          None      None   616.0   
618           2.023310e+13           44985.0          None      None   618.0   
646           2.023310e+13           44985.0          None      None   646.0   
650           2.023310e+13           44985.0          None      None   650.0   
667           2.023310e+13           44985.0          None      None   667.0   
747           2.023310e+13           44985.0          None      None   747.0   
755           2.023310e+13           44985.0          None      None   755.0   
766           2.023310e+13           44985.0          None      None   766.0   
775           2.023310e+13           44985.0          None      None   775.0   
787           2.023310e+13           44985.0          None      None   787.0   
810           2.023310e+13           44985.0          None      None   810.0   
832           2.023310e+13           44985.0          None      None   832.0   
836           2.023310e+13           44985.0          None      None   836.0   
865   

In [ ]:
print(duplicated_rows.shape)

(405, 77)


In [ ]:
# Assuming you have already created the 'result_df' DataFrame
duplicated_rows.to_excel('duplicated_rows.xlsx', index=False)  # Save to an Excel file

In [ ]:
from google.colab import files
files.download('duplicated_rows.xlsx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>